src/encoded/static/components/regulome_search.js

```diff
@@ -538,27 +538,29 @@ ResultsTable.defaultProps = {
     shortened: false,
 };
 
-// Sanitize user input and facet terms for comparison: convert to lowercase, remove white space and asterisks (which cause regular expression error)
-const sanitizedString = inputString => inputString.toLowerCase()
-    .replace(/ /g, '') // remove spaces (to allow multiple word searches)
-    .replace(/[*?()+[\]\\/]/g, ''); // remove certain special characters (these cause console errors)
-
-const FacetButton = (props) => {
-    const buttonLabel = props.buttonLabel;
-    const lookupFilterCount = props.lookupFilterCount;
-    const selectedFacets = props.selectedFacets;
-    const facetLabel = props.facetLabel;
-    const addGenomeFilter = props.addGenomeFilter;
-    return (
-        <button
-            className={selectedFacets.includes(`${buttonLabel}AND${facetLabel}`) ? 'active' : ''}
-            onClick={() => addGenomeFilter(buttonLabel, facetLabel)}
-            disabled={(lookupFilterCount(buttonLabel, facetLabel) === ' (0)')}
-        >
-            {buttonLabel}{lookupFilterCount(buttonLabel, facetLabel)}
-        </button>
-    );
-};
+class FacetButton extends React.Component {
+    constructor() {
+        super();
+        this._addGenomeFilter = this._addGenomeFilter.bind(this);
+    }
+
+    _addGenomeFilter() {
+        this.props.addGenomeFilter(this.props.buttonLabel, this.props.facetLabel);
+    }
+
+    render() {
+        const { buttonLabel, lookupFilterCount, selectedFacets, facetLabel } = this.props;
+        return (
+            <button
+                className={selectedFacets.includes(`${buttonLabel}AND${facetLabel}`) ? 'active' : ''}
+                onClick={this._addGenomeFilter}
+                disabled={(lookupFilterCount(buttonLabel, facetLabel) === ' (0)')}
+            >
+                {buttonLabel}{lookupFilterCount(buttonLabel, facetLabel)}
```

Comment:
The `lookupFilterCount` implementation doesn’t look entirely trivial, and it would get called with the same parameters twice in a row here. Maybe just call it once?